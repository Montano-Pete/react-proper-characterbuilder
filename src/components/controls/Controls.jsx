import React from 'react';

export default function Controls({
  head,
  onHeadChange,
  torso,
  onTorsoChange,
  pants,
  onPantChange,
  newCatchphrase,
  setCatchphrases,
  handleClick,
}) {
  const headChoices = ['bird', 'dog', 'duck'];
  const torsoChoices = ['blue', 'fancy', 'pink', 'red'];
  const pantChoices = ['blue', 'dog', 'leg', 'white'];

  return (
    <div>
      <label>
        Head Options
        <select
          value={head}
          onChange={(event) => onHeadChange(event.target.value)}
        >
          {headChoices.map((choice) => (
            <option key={choice}>{choice}</option>
          ))}
        </select>
      </label>

      <label>
        Torso Options
        <select
          value={torso}
          onChange={(event) => onTorsoChange(event.target.value)}
        >
          {torsoChoices.map((choice) => (
            <option key={choice}>{choice}</option>
          ))}
        </select>
      </label>

      <label>
        Pant Options
        <select
          value={pants}
          onChange={(event) => onPantChange(event.target.value)}
        >
          {pantChoices.map((choice) => (
            <option key={choice}>{choice}</option>
          ))}
        </select>
      </label>

      <label>
        Catch Phrase:
        <input
          type="text"
          value={newCatchphrase}
          onChange={(event) => setCatchphrases(event.target.value)}
        />
        <button type="button" onClick={handleClick}>
          Add a catchphrase!
        </button>
      </label>
    </div>
  );
}
