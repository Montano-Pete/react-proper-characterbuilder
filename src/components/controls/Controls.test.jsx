import { render, screen } from '@testing-library/react';
import Controls from './Controls';

it('should render Controls elements on page', () => {
  const { container } = render(<Controls />);

  const headSelector = screen.getByLabelText(/Head Options/);
  const torsoSelector = screen.getByLabelText(/Torso Options/);
  const pantSelector = screen.getByLabelText(/Pant Options/);

  expect(headSelector).toBeInTheDocument();
  expect(torsoSelector).toBeInTheDocument();
  expect(pantSelector).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
