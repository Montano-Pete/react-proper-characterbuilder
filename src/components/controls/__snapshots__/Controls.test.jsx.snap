// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Controls elements on page 1`] = `
<div>
  <div>
    <label>
      Head Options
      <select>
        <option>
          bird
        </option>
        <option>
          dog
        </option>
        <option>
          duck
        </option>
      </select>
    </label>
    <label>
      Torso Options
      <select>
        <option>
          blue
        </option>
        <option>
          fancy
        </option>
        <option>
          pink
        </option>
        <option>
          red
        </option>
      </select>
    </label>
    <label>
      Pant Options
      <select>
        <option>
          blue
        </option>
        <option>
          dog
        </option>
        <option>
          leg
        </option>
        <option>
          white
        </option>
      </select>
    </label>
    <label>
      Catch Phrase:
      <input
        type="text"
        value=""
      />
      <button
        type="button"
      >
        Add a catchphrase!
      </button>
    </label>
  </div>
</div>
`;
